temporal/t.rast.what/t.rast.what.py

```diff
@@ -460,84 +454,81 @@ def one_point_per_col_output(
     Each row represents a single raster map, hence a single time stamp
     """
     # open the output file for writing
-    out_file = open(output, "w") if output != "-" else sys.stdout
 
     first = True
-    for count in range(len(output_files)):
-        file_name = output_files[count]
-        gs.verbose(_("Transforming r.what output file %s" % (file_name)))
-        map_list = output_time_list[count]
-        in_file = open(file_name, "r")
-        lines = in_file.readlines()
-
-        matrix = []
-        for line in lines:
-            matrix.append(line.split(separator))
+    with open(output, "w") if output != "-" else sys.stdout as out_file:
```

Comment:
Python code:

```python
import sys
with sys.stdout as f:
    f.write("Hello\n")
sys.stdout.write("World\n")
```

produces:

```
Hello
Traceback (most recent call last):
  File "/tmp/test.py", line 4, in <module>
    sys.stdout.write("World\n")
ValueError: I/O operation on closed file.
```

And I think that will be unexpected for anyone who will touch this code later.

Reply:
It's not fixed yet, I only solved the conflicts

Reply:
I found a good solution:

```python
import sys
from contextlib import nullcontext

with nullcontext(sys.stdout) as f:
    f.write("Hello\n")
sys.stdout.write("World\n")

```

```python
vscode ➜ ~/grass (fix-resourcewarning-1) $ ipython
Python 3.12.7 (main, Nov  3 2024, 17:54:48) [GCC 11.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.30.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import sys
   ...: from contextlib import nullcontext
   ...: 
   ...: with nullcontext(sys.stdout) as f:
   ...:     f.write("Hello\n")
   ...: sys.stdout.write("World\n")
   ...: 
Hello
World
Out[1]: 6

In [2]: import sys
   ...: from contextlib import nullcontext
   ...: 
   ...: with nullcontext(sys.stdout) as f:
   ...:     f.write("Hello\n")
   ...: sys.stdout.write("World\n")
Hello
World
Out[2]: 6

In [3]: with nullcontext(sys.stdout) as f:
   ...:    ...:     f.write("Hello\n")
   ...: 
Hello

In [4]: sys.stdout.write("World\n")
World
Out[4]: 6

In [5]: 

```

Other solutions I've read was to use sys.stdout.fileno as the path inside open(), and also use closefd=False, like `with open(sys.stdout.fileno, "w", closefd=False) as out_file`, but it also meant having two "if"s inside the `open()` to conditionally set the file descriptor/path, and the closefd option. (Or it could've been set in variables in lines above too).

I'm opting to just "do nothing" with the nullcontext wrapper. Docs are here: https://docs.python.org/3/library/contextlib.html#contextlib.nullcontext
It is there for cases where you want a context manager, but conditionally.